src/api/general.c

```diff
@@ -46,9 +46,11 @@ static const char nlopt_algorithm_names[NLOPT_NUM_ALGORITHMS][256] = {
 #ifdef NLOPT_CXX
     "StoGO (global, derivative-based)",
     "StoGO with randomized search (global, derivative-based)",
+    "AGS (global, no-derivative)"
 #else
     "StoGO (NOT COMPILED)",
     "StoGO randomized (NOT COMPILED)",
+    "AGS (NOT COMPILED)"
```

Comment:
The missing `,` at the end of these items cause them to be merged with the item after which causes there to be one too few items in this array.